apis/logging/v1/input_source_types.go

```diff
@@ -0,0 +1,52 @@
+package v1
+
+import (
+	corev1 "k8s.io/api/core/v1"
+)
+
+// NOTE: The Enum validation on SourceSpec.Type must be updated if the list of types changes.
+
+// SourceSpec is a union of input source types.
+//
+// The fields of this struct define the set of known source types.
+type SourceSpec struct {
+	HTTP *HTTPSource `json:"http,omitempty"`
+}
+
+const (
+	FormatK8S  = "k8s"  // Log events in k8s list format, e.g. API audit log events.
+	FormatText = "text" // Newline-separated plain-text log records.
+)
+
+// HTTPSource receives encoded logs as a HTTP endpoint.
+type HTTPSource struct {
+
+	// FIXME TLS server configuration - can this be handled in the Service,
+	// or do we need vector to have certificates? Can the service be bypassed?
+
+	// FIXME What to do about service name clashes? Can't generate random name, service name is used to locate the source.
+
+	// Name of the source.
+	// A Service resource is generated for each source,
+	// the name must be a unique Service name in the ClusterLogForwarder's namespace.
+	//
+	// +required
+	Name string `json:"name"`
+
+	// Port that this source will listen on.
+	// Used to create a Service for this port.
+	//
+	// +required
+	Port corev1.ServicePort `json:"port"`
+	// Format is the format of incoming log data.
+	//
+	// +kubebuilder:validation:Enum:=k8s;text
+	// +required
+	Format string `json:"format"`
```

Comment:
Maybe we are limited by how we are integrating but could this not be determined from a ContentType header which will be more in line with typical http client/server interactions?

Reply:
Take a look at https://vector.dev/docs/reference/configuration/sources/http_server/#examples. Even though they are sending Content-Type on the wire, they're still specifying encoding in the config.  We would need to change vector to take encoding from Content-Type.

Reply:
Also, not everything is fully described by content-type. There is an explict content-type for ndjson which tells us everything, but if the content type is just "json" then it could be a k8s event list or some other JSON format.

Reply:
I'm not going to push the point if this is easier but we could define our own content types and what they represent.  I assume, however that our initial client won't be able to change so this is easier.

Reply:
We could use extended content types like `application/json+k8s-event`. The potential problem in a HTTP setting is confusion between the content type on incoming HTTP headers `application/json` and how it matches to our "extended" type. If we do use content-type in future, I'd prefer it to  "expected HTTP content-type header of incoming requests" and keep a separate "format" to mean "extra information about how to interpret the HTTP body". This is a preview release though, so we can re-consider this issue before we GA.